nautiluszim/scraper.py

```diff
@@ -342,29 +344,75 @@ def test_collection(self):
             all_names = zh.namelist()
 
         missing_files = []
+        saved_file_paths = []
         for entry in self.json_collection:
             if not entry.get("files"):
                 continue
-            for relative_path in entry["files"]:
-                if relative_path not in all_names:
-                    missing_files.append(relative_path)
+            for file in entry["files"]:
+                try:
+                    path = self.get_file_entry_from(file)[0]
+                    saved_file_paths.append(path)
+                    if not path.startswith("http") and path not in all_names:
+                        missing_files.append(path)
+                except ValueError:
+                    missing_files.append(entry["title"])
 
         if missing_files:
             raise ValueError(
                 "File(s) referenced in collection but missing:\n - "
                 + "\n - ".join(missing_files)
             )
+        duplicate_file_paths = [
+            path for path in saved_file_paths if saved_file_paths.count(path) > 1
+        ]
+        if duplicate_file_paths:
+            raise ValueError(
+                "Files in collection are duplicate:\n - "
+                + "\n - ".join(duplicate_file_paths)
+            )
+
+    def get_file_entry_from(
+        self, file: Union[str, Dict[str, str]]
+    ) -> Tuple[str | None, str | None]:
+        """Converting a file entity to the (path, filename)"""
+        # It's for old-format, pathname-only entries
+        if isinstance(file, str):
+            return (file, file)
+        archive_member = file.get("archive-member", None)
+        url = file.get("url", None)
+        path = None
+        filename = None
+        if archive_member is None and url is None:
```

Comment:
```suggestion
        if not archive_member and not url:
```

Not-None but empty should be treated equally here